src/main/java/org/opensearch/flowframework/template/WorkflowNode.java

```diff
@@ -0,0 +1,185 @@
+    /** The template field name for node inputs */
+    public static final String INPUTS_FIELD = "inputs";
+
+    private final String id; // unique id
+    private final String type; // maps to a WorkflowStep
+    private final Map<String, Object> inputs; // maps to WorkflowData
+
+    /**
+     * Create this node with the id and type, and any user input.
+     *
+     * @param id A unique string identifying this node
+     * @param type The type of {@link WorkflowStep} to create for the corresponding {@link ProcessNode}
+     * @param inputs Optional input to populate params in {@link WorkflowData}
+     */
+    public WorkflowNode(String id, String type, Map<String, Object> inputs) {
+        this.id = id;
+        this.type = type;
+        this.inputs = Map.copyOf(inputs);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        XContentBuilder xContentBuilder = builder.startObject();
+        xContentBuilder.field(ID_FIELD, this.id);
+        xContentBuilder.field(TYPE_FIELD, this.type);
+
+        xContentBuilder.startObject(INPUTS_FIELD);
+        for (Entry<String, Object> e : inputs.entrySet()) {
+            xContentBuilder.field(e.getKey());
+            if (e.getValue() instanceof String) {
+                xContentBuilder.value(e.getValue());
+            } else if (e.getValue() instanceof Map<?, ?>) {
+                Template.buildStringToStringMap(xContentBuilder, (Map<?, ?>) e.getValue());
+            } else if (e.getValue() instanceof Object[]) {
+                // This assumes an array of maps for "processor" key
+                xContentBuilder.startArray();
+                for (Map<?, ?> map : (Map<?, ?>[]) e.getValue()) {
+                    Template.buildStringToStringMap(xContentBuilder, map);
+                }
+                xContentBuilder.endArray();
+            }
+        }
+        xContentBuilder.endObject();
+
+        return xContentBuilder.endObject();
+    }
+
+    /**
+     * Parse raw json content into a workflow node instance.
+     *
+     * @param parser json based content parser
+     * @return the parsed WorkflowNode instance
+     * @throws IOException if content can't be parsed correctly
+     */
+    public static WorkflowNode parse(XContentParser parser) throws IOException {
+        String id = null;
+        String type = null;
+        Map<String, Object> inputs = new HashMap<>();
+
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            String fieldName = parser.currentName();
+            parser.nextToken();
+            switch (fieldName) {
+                case ID_FIELD:
+                    id = parser.text();
+                    break;
+                case TYPE_FIELD:
+                    type = parser.text();
+                    break;
+                case INPUTS_FIELD:
```

Comment:
Perhaps we can have handling specifically for pipeline configurations within this `INPUTS_FIELDS`. We can update the use case template to generalize pipeline/processor configurations like so:

```
"processors" : [
    {
    type : "<processor_type>",
    params : [
        "field1" : "value",
        "field2" : "value",
        "field3" : "value",
    ]
}
]
```

We can then map the processor type to some template and pass the params there to build the configuration

Reply:
Agreed, we just need a predictable way to identify when and where in the template tree we will use these.

The parser is written for the specific json committed with this PR, we can continue to update and iterate on that.

Reply:
Just had a thought, if this is part of another plugin, it might already have its NamedXContent loaded into opensearch and available for us to parse with.

Reply:
Yep, there's a `ProcessorInfo` class [that's already implemented this](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/ingest/ProcessorInfo.java)

Reply:
`ProcessorInfo` just gives us the list of processor types, not the configuration itself.